### PR TITLE
Log window implementation for self-education in templates and actions

### PR DESCRIPTION
**Pull request by jotelha**

### CODE REVIEW

File: dtool_lookup_gui/utils/logging.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    def max_lines(self, value: Optional[int] = None):
+        if value is None:
+            self._max_lines = None # no limit
+        elif isinstance(value, int) and value >= 0:
+            self._max_lines = value
+        else:
+            raise ValueError(f"max_lines must be None or int, not {type(value)}.")
+
+    def _insert_into_buffer(msg):
+        ...
```
Comment (pastewka) [inline dtool_lookup_gui/utils/logging.py:84]: Is this supposed to be an ellipsis?

Comment (jotelha) [reply dtool_lookup_gui/utils/logging.py:84]: Yes, that's there intentionally as an abstract method.

Comment (pastewka) [reply dtool_lookup_gui/utils/logging.py:84]: Okay, didn't know this syntax...

Comment (pastewka) [reply dtool_lookup_gui/utils/logging.py:84]: Okay, didn't know this syntax...

**Review by pastewka — APPROVED** [review]

**Pull request merged**